homeassistant/components/media_player/anthemav.py

```diff
@@ -0,0 +1,225 @@
+https://home-assistant.io/components/media_player.anthemav/
+"""
+import logging
+import asyncio
+
+import voluptuous as vol
+
+from homeassistant.components.media_player import (
+    PLATFORM_SCHEMA, SUPPORT_TURN_OFF, SUPPORT_TURN_ON, SUPPORT_SELECT_SOURCE,
+    SUPPORT_VOLUME_MUTE, SUPPORT_VOLUME_SET, MediaPlayerDevice)
+from homeassistant.const import (
+    CONF_NAME, CONF_HOST, CONF_PORT, STATE_OFF, STATE_ON, STATE_UNKNOWN,
+    EVENT_HOMEASSISTANT_STOP, CONF_SCAN_INTERVAL)
+import homeassistant.helpers.config_validation as cv
+
+REQUIREMENTS = ['anthemav>=1.1.3']
+
+_LOGGER = logging.getLogger(__name__)
+
+DOMAIN = 'anthemav'
+
+DEFAULT_NAME = "Anthem AVR"
+
+SUPPORT_ANTHEMAV = SUPPORT_VOLUME_SET | SUPPORT_VOLUME_MUTE | \
+    SUPPORT_TURN_ON | SUPPORT_TURN_OFF | SUPPORT_SELECT_SOURCE
+
+PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend({
+    vol.Optional(CONF_NAME, default=DEFAULT_NAME): cv.string,
+    vol.Required(CONF_HOST): cv.string,
+    vol.Required(CONF_PORT): cv.port,
+    vol.Optional(CONF_SCAN_INTERVAL): vol.All(vol.Coerce(int))
+    })
+
+SCAN_INTERVAL = 120
+
+
+@asyncio.coroutine
+def async_setup_platform(hass, config, async_add_devices, discovery_info=None):
+    """Set up our socket to the AVR."""
+    import anthemav
+
+    host = config.get(CONF_HOST)
+    port = config.get(CONF_PORT)
+
+    #
+    # If there's a better way to do this, I'm all ears.  I need to squash the
+    # global SCAN_INTERVAL value if the user has supplied an alternative scan
+    # interval in the configuration.  Since this is a local push/asyncio
+    # platform, I don't have direct control over the interval.  Setting this
+    # global is the only way I can tell hass itself how frequently to call
+    # async_update.
+    #
+    # pylint: disable=redefined-outer-name
+    # pylint: disable=invalid-name
+    SCAN_INTERVAL = config.get(CONF_SCAN_INTERVAL) or 120
+
+    _LOGGER.info('Provisioning Anthem AVR device at %s:%d scan every %d sec',
+                 host, port, SCAN_INTERVAL)
+
+    device = AnthemAVR(hass, host, port)
+
+    yield from async_add_devices([device])
+
+    def anthemav_update_callback(message):
+        """Receive notification from transport that new data exists."""
+        _LOGGER.info('Received update callback from AVR: %s', message)
+        hass.async_add_job(device.async_update_ha_state)
+
+    avr = yield from anthemav.Connection.create(
+        host=host, port=port, loop=hass.loop,
+        update_callback=anthemav_update_callback)
+
+    device.avr = avr
+
+    _LOGGER.debug('dump_devicedata: '+device.dump_avrdata)
+    _LOGGER.debug('dump_conndata: '+avr.dump_conndata)
+    _LOGGER.debug('dump_rawdata: '+avr.protocol.dump_rawdata)
+
+    hass.bus.async_listen_once(EVENT_HOMEASSISTANT_STOP, device.avr.close)
+
+
+class AnthemAVR(MediaPlayerDevice):
+    """Entity reading values from Anthem AVR protocol."""
+
+    def __init__(self, hass, host, port):
```

Comment:
Remove `host` and `port` from the constructor and the corresponding attributes if they won't be used.